timeouts/datasource/schema.go

```diff
@@ -99,3 +112,25 @@ func attributesMap(opts Opts) map[string]schema.Attribute {
 
 	return attributes
 }
+
+func attrTypesMap(opts Opts) map[string]attr.Type {
+	attrTypes := map[string]attr.Type{}
+
+	if opts.Create {
+		attrTypes[attributeNameCreate] = types.StringType
+	}
```

Comment:
Not due to this pull request, but data source schemas should only support a `read` timeout.